pitest/src/main/java/org/pitest/classpath/CodeSource.java

```diff
@@ -57,7 +63,9 @@ public CodeSource(final ProjectClassPaths classPath,
   public List<ClassInfo> getTests() {
     return flatMap(this.classPath.test(), nameToClassInfo()).filter(
         and(isWithinATestClass(), isIncludedClass(),
-            not(ClassInfo.matchIfAbstract())));
+            not(ClassInfo.matchIfAbstract()),
+            not(ClassInfo.matchIfAnnotationPresentWithValueString(RunWith.class, 
```

Comment:
I don't think it's a good idea to check this in CodeSource. I think we should move that to JUnitTestClassIdentifier.isIncludedClass()

This is JUnit specific behaviour and shouldn't be executed in TestNG setups. I'm surprised that it seems to run inside a TestNG setup. But Annotations are usually treated differently by the class loader.